### PR TITLE
Fix: appKeys handling

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes issue with editing app keys through admin/apps

**Environment**: Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Create a new app that has requirement of keys and try to edit keys

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/trpc/server/routers/viewer/apps.tsx [hunk 1]
```diff
@@ -255,9 +255,7 @@ export const appsRouter = router({
       })
     )
     .mutation(async ({ ctx, input }) => {
-      let appKey = deriveAppDictKeyFromType(input.type, appKeysSchemas);
-      if (!appKey) appKey = deriveAppDictKeyFromType(input.slug, appKeysSchemas);
-      const keysSchema = appKeysSchemas[appKey as keyof typeof appKeysSchemas];
+      const keysSchema = appKeysSchemas[input.slug as keyof typeof appKeysSchemas];
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/apps.tsx:258]: We are already using dirName for this https://github.com/calcom/cal.com/blob/55a4510167db026686768dec4058ff971a9d8747/packages/trpc/server/routers/viewer/apps.tsx#L66. So made it consistent.

It was causing keys to be not saved for any new app

File: packages/app-store/appStoreMetaData.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export const appStoreMetadata = {} as AppStoreMetaData;
 for (const [key, value] of Object.entries(rawAppStoreMetadata)) {
   appStoreMetadata[key as keyof typeof appStoreMetadata] = {
     appData: null,
+    dirName: "dirName" in value ? value.dirName : value.slug,
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply packages/app-store/appStoreMetaData.ts:15]: Without this any new app that has requirement of keys, was not showing Keys Edit icon

**Comment by joeauyeung** [comment]
@hariombalhara I tried testing, but I can't save any keys. Do you get this error?

image

**Comment by hariombalhara** [comment]
I will recheck. Didn't see this error

**Comment by hariombalhara** [comment]
@joeauyeung Thanks for catching the error !! All fixed now.

**Review by zomars — APPROVED** [review]
Nice job @hariombalhara and @joeauyeung  🙏🏽

**Pull request merged**